app/views/ReferenceSelectView/ReferenceSelectView.jsx

```diff
@@ -195,25 +169,34 @@ class ReferenceSelectView extends React.Component {
   }
 
   render() {
-    const {onBuildBaseUrl, route, state} = this.props
-    const {api} = state.app.config
     const {
-      collection: collectionName,
-      documentId,
-      page,
-      referenceField: referenceFieldName
-    } = route.params
+      collection,
+      contentKey,
+      isNewDocument,
+      isSingleDocument,
+      onBuildBaseUrl,
+      parentContentKey,
+      route,
+      selectionKey,
+      state
+    } = this.props
+    const {documentId, page, referenceField: referenceFieldName} = route.params
     const {search} = route
     const parsedPage = Number.parseInt(page)
     const pageNumber = parsedPage.toString() === page ? parsedPage : undefined
 
-    // Getting the schema for the parent collection.
-    const collection =
-      collectionName === Constants.MEDIA_COLLECTION_SCHEMA.slug
-        ? Constants.MEDIA_COLLECTION_SCHEMA
-        : api.collections.find(collection => {
-            return collection.slug === route.params.collection
-          })
+    if (!(isNewDocument || isSingleDocument || documentId)) {
```

Comment:
```suggestion
    if (!isNewDocument && !isSingleDocument && !documentId)) {
```

... is easier on the brain, I think?